test/integration/multitenant_enforcement_test.rb

```diff
@@ -54,13 +55,33 @@ class MultitenantEnforcementTest < ActionDispatch::IntegrationTest
     service.update_column(:tenant_id, @provider.tenant_id)
     plan.update_column(:tenant_id, @provider.tenant_id + 1)
     token = FactoryBot.create(:access_token, owner: master_account.admin_users.first!, scopes: %w[account_management]).value
-    AccessToken.new.value
     get admin_api_service_application_plans_path(service_id: service.id, format: :json, access_token: token)
     assert_response :success
     put admin_api_service_application_plan_path(service_id: service.id, id: plan.id, format: :json), params: {access_token: token, description: "desc1"}
     assert_response :success
   end
 
+  test "multitenant master can retrieve from multiple tenants by http basic auth" do
+    host! master_account.external_admin_domain
+    service = master_account.first_service!
+    plan = FactoryBot.create(:application_plan, issuer: service)
+    service.update_column(:tenant_id, @provider.tenant_id)
+    plan.update_column(:tenant_id, @provider.tenant_id + 1)
+
+    token = FactoryBot.create(:access_token, owner: master_account.admin_users.first!, scopes: %w[account_management]).value
+    auth_pair = []
+    auth_pair << ["", token]
+    auth_pair << [token, ""]
+    auth_pair << ["", master_account.provider_key]
+    auth_pair << [master_account.provider_key, ""]
+
+    auth_pair.each do |pair|
+      basic_auth_str = ActionController::HttpAuthentication::Basic.encode_credentials(*pair)
+      get admin_api_service_application_plans_path(service_id: service.id, format: :json), headers: { 'HTTP_AUTHORIZATION' => basic_auth_str }
+      assert_response :success
+    end
+  end
```

Comment:
I think I need an explanation for this scenario. So there is a plan belonging to a service but with different `tenant_id`. That looks wrong, why must it be allowed when the request is authenticated with HTTP basic auth?

Reply:
> **multitenant master** can retrieve from multiple tenants by http basic auth

There are other scenarios that are the same except "by provider key", "by token", "by api_key".

Master must be allowed to see anything.